src/main/scala/Domain/Door.scala

```diff
@@ -56,7 +56,7 @@ object Door {
       }
 
   val aggregateLogic: List[Flow[Unit]] = List(
-    handler { case Register(id) => emitEvent(Registered(id)) } >> waitFor { case Registered(_) => () },
+    handler { handler[Register, Registered] } >> waitFor { case Registered(_) => () },
```

Comment:
So far I like any of promote or direct the most. Promote maybe is easier to reason about :) 

I think pass would be more fitting for this function combined with the new idea I wrote in the blog comment - to also combine an event handler with a continuation here - but it's a bit more work so can come later :)